kore/src/Kore/Rewrite/Rule/Simplify.hs

```diff
@@ -119,28 +116,23 @@ simplifyClaimRule ::
     MonadSimplify simplifier =>
     ClaimPattern ->
     simplifier (MultiAnd ClaimPattern)
-simplifyClaimRule =
-    fmap MultiAnd.make . Logic.observeAllT . worker
-  where
-    simplify
-        , filterWithSolver ::
-            Pattern RewritingVariableName ->
-            LogicT simplifier (Pattern RewritingVariableName)
-    simplify =
-        (return . Pattern.requireDefined)
-            >=> Pattern.simplifyTopConfiguration
-            >=> Logic.scatter
-            >=> filterWithSolver
-    filterWithSolver = SMT.Evaluator.filterBranch
-
-    worker :: ClaimPattern -> LogicT simplifier ClaimPattern
-    worker claimPattern = do
-        let lhs = ClaimPattern.left claimPattern
-        simplified <- simplify lhs
+simplifyClaimRule claimPattern = fmap MultiAnd.make $
```

Comment:
It seemed like `worker` is better suited as the body of the `simplifyClaimRule` function rather than in a where clause. When I see `make . observeAllT . worker` as the function body I still have to examine the `worker` function to understand what is happening. Worker is not a descriptive function name for the underlying action so I think it is easier to read when it is the body. I first tried thinking of a more descriptive name than worker, but the best I could think of was the same as simplifyClaimRule so I decided to move it up as the function body.